active-record/src/test/java/com/iluwatar/activeobject/ActiveRowTest.java

```diff
@@ -0,0 +1,56 @@
+package com.iluwatar.activeobject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+
+public class ActiveRowTest {
+
+	@Test
+	void initialiseTest() {
+		ActiveDatabase activeDatabase = new ActiveDatabase("world", "root", "apple-trunks", "localhost:3306", "city");
+		ActiveRow row = new ActiveRow(activeDatabase, "1");
+		assertDoesNotThrow(row::initialise);
+	}
+
+
+	@Test
+	void readRowTest() {
+		ActiveDatabase activeDatabase = new ActiveDatabase("world", "root", "apple-trunks", "localhost:3306", "city");
+		ActiveRow row = new ActiveRow(activeDatabase, "1");
+		System.out.println(row.read().size());
+		int rowSize = row.read().size();
+		if (!(row.read().equals(new ArrayList<String>()))) {
+			for (int i = 0; i < rowSize; i++) {
+				assertNotNull(row.read().get(i));
+			}
+		}
+
+	}
+
+	@Test
+	void deleteRowTest() {
+		ActiveDatabase activeDatabase = new ActiveDatabase("world", "root", "apple-trunks", "localhost:3306", "city");
+		ActiveRow row = new ActiveRow(activeDatabase, "1");
+		row.delete();
+		assertEquals(row.read(), new ArrayList<String>());
+
+	}
+
+	@Test
+	void insertRowTest() {
+		ActiveDatabase activeDatabase = new ActiveDatabase("world", "root", "apple-trunks", "localhost:3306", "city");
+		ActiveRow row = new ActiveRow(activeDatabase, "1");
+		row.delete();
+		row.contents = new ArrayList<String>(
+				Arrays.asList("101", "Godoy Cruz", "ARG", "Mendoza", "206998"));
+		assertEquals(row.read(), Arrays.asList("101", "Godoy Cruz", "ARG", "Mendoza", "206998"));
```

Comment:
You delete the row and then assign it's contents, so I don't think it's fair to call it the insertRowTest. What would deleted rows 29-50? Isn't that the test you want?

Reply:
Good catch, yep I meant to row.delete() then update contents, then write it and then read().